exec/java-exec/src/main/java/org/apache/drill/exec/server/options/FallbackOptionManager.java

```diff
@@ -83,14 +84,30 @@ public OptionValue getOption(final String name) {
    */
   abstract boolean setLocalOption(OptionValue value);
 
+  /**
+   * Deletes the option with given name for this manager without falling back.
+   *
+   * @param type option type
+   * @return true iff the option was successfully deleted
```

Comment:
this javadoc can have multiple meanings: _successfully deleted_ could either mean the implementation couldn't find the option, or it doesn't support the option type. It should actually return false if it doesn't support the option type